README.md

```diff
@@ -179,6 +179,10 @@ A Linux or MacOS environment is required. Windows is not supported and the npm c
 3. run `npm run dev` to start the webserver, open up
    http://localhost:3001/#search and try out the basic features
 
+Before merging a branch, create a commit that updates the package's version number in `package.json`.
+Also, update the version in `package-lock.json`, either manually or through NPM (e.g. `npm i`).
```

Comment:
Why would package-lock.json ever need to be updated manually? I thought that's updated automatically by the build process?

Reply:
What happens if someone forgets this step and the versions in package.json and package-lock.json differ?

Reply:
Changes by the TC build pipeline are not persisted in the repo. If you don't care about an out of sync package-lock, we can ignore this step. It could be a minor source of confusion to somebody looking at package and package-lock files, but there no technical consequences.

Reply:
So TC runs `npm ci`, which shouldn't update package-lock.json anyway, as I understand. So I guess we could say something like

> Before creating a PR:
> - update the version in `package.json`, preferably in a separate commit
> - run `npm i` to update `package-lock.json` and commit the changes

But then do we want a separate commit for the `package-lock.json` too? Those 2 updates can be done in one commit, right? I.e. you do all your "substantial" changes, commit, update version, `npm i`, commit?

Reply:
Yes, the version updates should go into the same commit. The steps you mention are exactly what I am thinking.

For the current workload this workflow should work. But once there are multiple parallel PRs, they will compete for the version update. It also blocks publishing multiple changes in a single version increment.
Maybe a future topic to tackle.

Reply:
Yes, I think that's fine. In the rare case of multiple parallel PRs we can just have only one of them update the version number, and that one is the last to be merged. Merging the other PRs (without a version number update) will simply fail to publish the npm package, since the version already exists, right?